[DKP] Release Notes 2.2.3 (#4636)
mesosphere/dcos-docs-site: pages/dkp/kommander/2.2/release-notes/2.2.3/index.md

---
layout: layout.pug
navigationTitle: DKP 2.2.3 Release Notes
title: DKP 2.2.3 Release Notes
menuWeight: 30
excerpt: View release-specific information for DKP 2.2.3
enterprise: false
beta: false
---
**D2iQ&reg; Kommander&reg; version 2.2.3 was released on October 14th, 2022.**

[button color="purple" href="https://support.d2iq.com/hc/en-us/articles/4409215222932-Product-Downloads"]Download DKP[/button]

<p class="message--note"><strong>NOTE: </strong>You must be a registered user and logged on to the support portal to download this product. New customers must contact their sales representative or <a href="mailto:sales@d2iq.com">sales@d2iq.com</a> before attempting to download or install DKP.</p>

## Release summary

> Welcome to D2iQ Kubernetes Platform (DKP) 2.2.3! This release provides new features and enhancements to improve the user experience, fix reported issues, integrate changes from previous releases, and maintains compatibility and support for other packages used in Kommander.

Reviewer: ```suggestion
Welcome to D2iQ Kubernetes Platform (DKP) 2.2.3! This release fixes reported issues, integrates changes from previous releases, and maintains compatibility and support for other packages used in Kommander.
```

DKP 2.2.x supports Kubernetes versions between 1.21.0 and 1.22.x. Any cluster you want to attach using DKP 2.2.x must be running a Kubernetes version in this range.

### Supported versions

| Kubernetes Support | Version |
| ------------------ | ------- |
| **Minimum**  | 1.21.0  |
| **Maximum**  | 1.22.x  |
| **Default**  | 1.22.8  |

## Fixes and Improvements

> ### Chartmuseum not hosting with the cert provided to Kommander during installation, x509 unknown error on Workload cluster (D2IQ-93002)

Reviewer 2: ```suggestion
### ChartMuseum not hosting with the cert provided to Kommander during installation, x509 unknown error on Workload cluster (D2IQ-93002)
```

> ### Chartmuseum not hosting with the cert provided to Kommander during installation, x509 unknown error on Workload cluster (D2IQ-93002)
> 
> [Looking at the ticket, do we have a fix for this? We've resolved CA certs below but not the connected missing airgap bundle ticket.]

Reviewer 3: The JIRA ticket summary is kind of hard to understand, and there's actually two issues here (both are resolved, I think the bundle one is just waiting for a release to close out)

```suggestion
### Workload clusters cannot be successfully attached when the management cluster uses a custom domain and certificate (D2IQ-93002)

A problem that caused the kommander federation-controller to use system certificates instead of the configured custom certificates was corrected.   The federation-controller now uses custom certificates if they are present.  

### Missing Cert-manager images in airgapped bundles (D2IQ-93002)

The air-gapped image bundles did not include images for cert-manager, which prevented successful deployment of the platform applications to managed and attached clusters in those environments.   The bundle has been updated to include the correct images.
```

### Cannot resolve CA certificate when provided as custom certificate (D2IQ-91798)

To resolve this issue, use a custom image for the Kommander federation controllers that includes the patch. The easiest way to do that is by using the installer config. This works both when:

* Installing a new cluster

* Re-running `dkp install kommander` on an existing v2.2.2 cluster

In the installer configuration, under apps, add the following lines:
```
  kommander:
    enabled: true
    values: |
      controller:
        containers:
          manager:
            image:
              repository: mesosphere/kommander2-federation-controller-manager
              tag: <patched_image_tag>

> ```

Reviewer 2: Is this really how it would work anymore @cbuto @cwyl02 @gracedo ?
If they download the new DKP CLI for 2.2.3, and then run
`dkp install kommander`
or, upgrade, or whatever it is that they would want, 
wouldn't this come with that?

This _would_ work, but not sure if that's easier than just running install from the new dkp CLI file, etc., because we would have to tell them this tag that's in a private section of Docker Hub, etc.

These are the "workaround" steps, compared to the preferred workflow.

Reviewer 3: @msdolbey you are correct, these are the 'hotfix' instructions.  The 2.2.3 release includes the necessary fix (see above) and so we don't need the the hotfix instructions.  It will just work properly now.

```suggestion
```

### No apparent upgrade path for 1.X vSphere clusters to 2.X (D2IQ-90769)

> vSphere clusters can now be migrated from 1.X to 2.X versions of DKP.

Reviewer 3: This issue is 'fixed' in this release, so at a minimum this should be removed.  I am not sure though how useful it is to document this issue in 2.2.x because the upgrade path for 1.8 clusters requires upgrading first to 2.1.x;  you can then upgrade to 2.2.   

The problem described in this 'known issue' occurs when upgrading to 2.1.x (and is fixed in 2.1.4), so it would never be encountered in a '2.2' environment.

@dkoshkin  - what say you?

Author: This section was simply removed @cneth

## Component and Application updates

When upgrading to this release, the following services and service components are upgraded to the listed version:

### Components

> - CAPPPVersion = "v0.8.3"

Reviewer: Do these need updating? @bcmendoza can help.

Reviewer 3: Bump to v0.8.4:  ref: https://github.com/mesosphere/konvoy2/pull/1617

```suggestion
- CAPPPVersion = "v0.8.4"
```

- CAPZVersion  = "v1.3.1"
- Containerd 1.4.13
- KIB v1.17.2

> ### Applications

Reviewer: Do these need updating?

Reviewer 4: @mwrightd2 you can update this section with the latest version of the automated table here: https://github.com/mesosphere/dkp-infra-tools/blob/6c64a8dda91cd9865e8084e8c9eab802e7b3cf78/automation/cmd/report/hack/v2.2.3-rc.4.md?plain=1

Author: Updated @gracedo
